### PR TITLE
Switch to v2 endpoints

### LINKED ISSUE #57: Switch to the endpoints of the new coverage backend

### PR DESCRIPTION
**Pull request by La0**
Fixes #57

### CODE REVIEW

**Comment by La0** [comment]
@marco-c We should copy `/coverage/supported_extensions` to `/v2/extensions`, so we can kill all `/coverage` endpoints in the end

**Comment by La0** [comment]
`/v2/extensions` done in https://github.com/mozilla/release-services/pull/2151

File: build.js [hunk 1]
```diff
@@ -25,9 +26,11 @@ fs.readFile('manifest.json', 'utf8', function(err, data) {
   });
 });
 
-https.get('https://coverage.moz.tools/coverage/supported_extensions', res => {
+https.get(config.BACKEND_URL + '/v2/extensions', res => {
+  if(res.statusCode !== 200)
```
Comment (marco-c) [inline build.js]: Nit: `if (` and `{` `}`

File: build.js [hunk 2]
```diff
@@ -25,9 +26,11 @@ fs.readFile('manifest.json', 'utf8', function(err, data) {
...
+    throw new Error('Invalid response from backend: ' + res.statusCode)
+
   let data = '';
-
```
Comment (marco-c) [inline build.js:30]: Nit: don't remove newline

File: coverage.js [hunk 3]
```diff
@@ -4,8 +4,13 @@
 
 'use strict';
 
+const config = require('./config.js');
+
+
 async function fetchCoverage(rev, path) {
-  let response = await fetch(`https://coverage.moz.tools/coverage/file?changeset=${rev}&path=${path}`);
+  console.debug('Fetch Coverage for', rev, path);
+  // TODO: add changeset argument when it works with latest rev
```
Comment (marco-c) [inline coverage.js]: ?

**Review by marco-c — APPROVED** [review]
Did you test it with both Searchfox and DXR?

File: build.js [hunk 4]
```diff
@@ -25,9 +27,14 @@ fs.readFile('manifest.json', 'utf8', function(err, data) {
   });
 });
 
-https.get('https://coverage.moz.tools/coverage/supported_extensions', res => {
+console.log('Using backend:', CONFIG.BACKEND_URL);
+https.get(CONFIG.BACKEND_URL + '/v2/extensions', res => {
+  if (res.statusCode !== 200) {
+    throw new Error('Invalid response from backend: ' + res.statusCode);
+	}
```
Comment (marco-c) [inline build.js]: Indentation is off here

File: config.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+// Old style exports
+window.CONFIG = {
+	'BACKEND_URL' : 'https://coverage.moz.tools',
```
Comment (marco-c) [inline config.js]: Intendation is a tab instead of two spaces here

File: coverage.js [hunk 6]
```diff
@@ -5,7 +5,8 @@
 'use strict';
 
 async function fetchCoverage(rev, path) {
-  let response = await fetch(`https://coverage.moz.tools/coverage/file?changeset=${rev}&path=${path}`);
+  console.debug('Fetch Coverage for', rev, path);
+  let response = await fetch(CONFIG.BACKEND_URL + `/v2/path?path=${path}&changeset=${rev}`);
```
Comment (marco-c) [inline coverage.js]: Nit: \`${CONFIG.BACKEND_URL}/v2/...\`

File: manifest.json [hunk 7]
```diff
@@ -10,32 +10,32 @@
...
                   "*://hg.mozilla.org/mozilla-central/annotate/*"],
-      "js": ["supported_extensions.js", "coverage.js", "button.js", "hgmo.js"],
+      "js": ["supported_extensions.js", "config.js", "coverage.js", "button.js", "hgmo.js"],
       "css": ["spinner.css", "dxr.css"],
       "run_at": "document_end"
     }
   ],
   "permissions": [
-    "https://coverage.moz.tools/coverage/*",
+    "https://coverage.testing.moz.tools/v2/*",
```
Comment (marco-c) [inline manifest.json]: Need to update the permissions here

**Comment by La0** [comment]
Current version works on DXR + searchfox : https://imgur.com/a/f3CU7VO

File: build.js [hunk 8]
```diff
@@ -25,9 +27,13 @@ fs.readFile('manifest.json', 'utf8', function(err, data) {
   });
 });
 
-https.get('https://coverage.moz.tools/coverage/supported_extensions', res => {
+https.get(CONFIG.BACKEND_URL + '/v2/extensions', res => {
```
Comment (marco-c) [inline build.js:30]: Nit: you could use a template string here too

**Review by marco-c — APPROVED** [review]

**Pull request merged**